src/unix/linux_like/linux/musl/mod.rs

```diff
@@ -457,31 +442,26 @@ s_no_extra_traits! {
         pub ut_host: [c_char; 256],
         pub ut_exit: __exit_status,
 
-        #[cfg(target_env = "musl")]
-        #[cfg(not(target_arch = "loongarch64"))]
+        // loongarch64 and ohos already use the new ABI
+        #[cfg(not(any(target_arch = "loongarch64", target_env = "ohos", musl_v1_2_3)))]
+        #[deprecated(
+            since = "0.2.173",
+            note = "The ABI of this field has changed from c_long to c_int with padding, \
+                we'll follow that change in the future release as part of an effort to \
+                update the musl API to reflect upstream. See #4443 for more info."
+        )]
         pub ut_session: c_long,
 
-        #[cfg(target_env = "musl")]
-        #[cfg(target_arch = "loongarch64")]
-        pub ut_session: c_int,
-
-        #[cfg(target_env = "musl")]
-        #[cfg(target_arch = "loongarch64")]
+        #[cfg(any(target_arch = "loongarch64", target_env = "ohos", musl_v1_2_3))]
```

Comment:
Maybe it's worth setting `musl_v1_2_3` via build.rs if the target is loongarch64? Since we only support/test 1.2.3+ on that target.

ohos might fall into this same boat based on config.

Reply:
Yeah, this is definitely a lot cleaner. I was thinking about the time64 changes, which are not on loongarch64 yet - but these can be controlled by a separate cfg, which can only be enabled through env when `musl_v1_2_3` is.

Reply:
I'm not sure I follow - what changes to time are there on loongarch64 since that is a 64-bit platform? And what would this separate config be?